mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -395,12 +397,16 @@ func (b *builder) deleteCacheController(clusterID string) error {
 	return nil
 }
 
+var clusterNsEnv, _ = env.GetVar("POD_NAMESPACE", "", "")
+
 func initMultiClusterSecretController(b *builder, kubeconfig string, env adapter.Env) (err error) {
 	var clusterNs string
 
 	paramsProto := b.adapterConfig
 	if clusterNs = paramsProto.ClusterRegistriesNamespace; clusterNs == "" {
-		if clusterNs = os.Getenv("POD_NAMESPACE"); clusterNs == "" {
+		if clusterNsEnv != "" {
+			clusterNs = clusterNsEnv
+		} else {
 			clusterNs = defaultClusterRegistriesNamespace
```

Comment:
Unless you are trying to minimize code changes I think you could just set this as the default value?

Reply:
Done.